client/me/privacy/main.jsx

```diff
@@ -135,9 +156,73 @@ const Privacy = createReactClass( {
 						</FormButton>
 					</form>
 				</Card>
+				<SectionHeader label={ translate( 'Legal' ) } />
+				<Card>
+					<p>
+						{ translate(
+							'A Data Processing Addendum (DPA) allows web sites and companies to assure customers, vendors, ' +
+								'and partners that their data handling complies with the law.'
+						) }
+					</p>
+
+					<p>{ translate( 'Note: most free site owners or hobbyists do not need a DPA.' ) }</p>
+
+					<p>
+						<strong>
+							{ translate(
+								'Having a DPA does not change any of our privacy and security practices for site visitors. ' +
+									'Everyone using our service gets the same high standards of privacy and security.'
+							) }
+						</strong>
+					</p>
+					<Button primary className="privacy__dpa-request-button" onClick={ this.props.requestDpa }>
+						{ translate( 'Request a DPA' ) }
+					</Button>
+				</Card>
 			</Main>
 		);
 	},
 } );
 
-export default compose( localize, protectForm )( Privacy );
+const dpaRequestId = 'dpa-request';
+function requestDpa() {
+	requestHttpData(
+		dpaRequestId,
+		http( {
+			apiNamespace: 'wpcom/v2',
+			method: 'POST',
+			path: '/me/request-dpa',
```

Comment:
this may seem bizarre, and you don't need to run this experiment, but you might be able to skip the `componentDidUpdate()` function altogether by passing the `onSuccess` and `onFailure` props to `http()` here. it was designed to call these actions in response to network activity and they may be replaced by the HTTP middleware and such but possibly could leave the global notices as a side-effect of the API call and not of the UI

```ts
http( {
    …
    onSuccess: successNotice( translate( '…' ) ),
    onFailure: errorNotice( translate( '…' ) )
    …
} );
```

the drawback obviously is that you don't get access to the response data to inform the error message and possibly this won't work at all

Reply:
That's good to know.